assets/js/stripe.js

```diff
@@ -398,305 +508,281 @@ jQuery( function( $ ) {
-					if ( 'yes' === wc_stripe_params.is_checkout ) {
-						return true;
-					} else {
-						wc_stripe_form.openModal();
-						return false;
-					}
+			// Open the Stripe Checkout modal.
+			if ( 'yes' === wc_stripe_params.is_stripe_checkout && wc_stripe_form.isStripeModalNeeded() && wc_stripe_form.isStripeCardChosen() ) {
+				if ( 'yes' === wc_stripe_params.is_checkout ) {
+					return true;
+				} else {
+					wc_stripe_form.openModal();
+					return false;
 				}
+			}
 
-				/*
-				 * For methods that needs redirect, we will create the
-				 * source server side so we can obtain the order ID.
-				 */
-				if (
-					wc_stripe_form.isBancontactChosen() ||
-					wc_stripe_form.isGiropayChosen() ||
-					wc_stripe_form.isIdealChosen() ||
-					wc_stripe_form.isAlipayChosen() ||
-					wc_stripe_form.isSofortChosen() ||
-					wc_stripe_form.isP24Chosen() ||
-					wc_stripe_form.isEpsChosen() ||
-					wc_stripe_form.isMultibancoChosen()
-				) {
-					if ( $( 'form#order_review' ).length ) {
-						$( 'form#order_review' )
-							.off(
-								'submit',
-								this.onSubmit
-							);
-
-						wc_stripe_form.form.submit();
-
-						return false;
-					}
-
-					if ( $( 'form.woocommerce-checkout' ).length ) {
-						return true;
-					}
-
-					if ( $( 'form#add_payment_method' ).length ) {
-						$( 'form#add_payment_method' )
-							.off(
-								'submit',
-								this.onSubmit
-							);
+			// For methods that needs redirect, we will create the source server side so we can obtain the order ID.
+			if (
+				wc_stripe_form.isBancontactChosen() ||
+				wc_stripe_form.isGiropayChosen() ||
+				wc_stripe_form.isIdealChosen() ||
+				wc_stripe_form.isAlipayChosen() ||
+				wc_stripe_form.isSofortChosen() ||
+				wc_stripe_form.isP24Chosen() ||
+				wc_stripe_form.isEpsChosen() ||
+				wc_stripe_form.isMultibancoChosen()
+			) {
+				if ( $( 'form#order_review' ).length ) {
+					$( 'form#order_review' )
+						.off(
+							'submit',
+							this.onSubmit
+						);
 
-						wc_stripe_form.form.submit();
+					wc_stripe_form.form.submit();
 
-						return false;
-					}
+					return false;
 				}
 
-				wc_stripe_form.createSource();
+				if ( $( 'form#add_payment_method' ).length ) {
+					$( 'form#add_payment_method' )
+						.off(
+							'submit',
+							this.onSubmit
+						);
 
-				// Prevent form submitting
-				return false;
-			} else if ( $( 'form#add_payment_method' ).length ) {
-				e.preventDefault();
-
-				// Stripe Checkout.
-				if ( 'yes' === wc_stripe_params.is_stripe_checkout && wc_stripe_form.isStripeModalNeeded() && wc_stripe_form.isStripeCardChosen() ) {
-					wc_stripe_form.openModal();
+					wc_stripe_form.form.submit();
 
 					return false;
 				}
 
-				wc_stripe_form.block();
-
-				wc_stripe_form.createSource();
-				return false;
+				return true; // The checkout form will be sumitted if this is !== false.
```

Comment:
This will be better as a function comment, instead of explaining just this one "return".